server/models/pattern/core/pattern.go

```diff
@@ -207,6 +208,26 @@ func NewPatternFile(yml []byte) (af Pattern, err error) {
 	return
 }
 
+// isValidPattern checks if the pattern file is valid or not
+func IsValidPattern(stringifiedFile string) (err error) {
+	pattern := Pattern{}
+
+	if err = yaml.Unmarshal([]byte(stringifiedFile), &pattern); err != nil {
+		return err
+	}
+
+	if pattern.Services == nil {
+		return errors.New("missing services field")
+	}
+
+	for serviceName, service := range pattern.Services {
+		if service.Traits == nil {
+			return errors.New("missing traits field for:" + serviceName)
+		}
+	}
```

Comment:
Not required for a pattern/design to always have traits, although we have traits for every pattern/design now, this is not an important check.